recipes/fftw/all/conanfile.py

```diff
@@ -70,64 +87,93 @@ def generate(self):
         tc.variables["ENABLE_OPENMP"] = self.options.openmp
         tc.variables["ENABLE_THREADS"] = self.options.threads
         tc.variables["WITH_COMBINED_THREADS"] = self.options.get_safe("combinedthreads", False)
-        tc.variables["ENABLE_FLOAT"] = self.options.precision == "single"
-        tc.variables["ENABLE_LONG_DOUBLE"] = self.options.precision == "longdouble"
         tc.variables["ENABLE_SSE"] = self.options.simd == "sse"
         tc.variables["ENABLE_SSE2"] = self.options.simd == "sse2"
         tc.variables["ENABLE_AVX"] = self.options.simd == "avx"
         tc.variables["ENABLE_AVX2"] = self.options.simd == "avx2"
         tc.generate()
 
+    @property
+    def build_folder(self):
+        bf = super().build_folder
+        return os.path.join(bf, self._current_precision) if self._current_precision else bf
+
+    @property
+    def _all_precisions(self):
+        return [p for p in ALL if self.options.precision == p or self.options.get_safe(f"precision_{p}")]
+
     def build(self):
+        def on_off(value):
+            return "ON" if value else 'OFF'
+
         apply_conandata_patches(self)
-        cmake = CMake(self)
-        cmake.configure()
-        cmake.build()
+        for self._current_precision in self._all_precisions:
+            cmake = CMake(self)
+            cli_args = [
+                f"-DENABLE_FLOAT={on_off(self._current_precision == SINGLE)}",
+                f"-DENABLE_LONG_DOUBLE={on_off(self._current_precision == LONGDOUBLE)}",
+                f"-DENABLE_QUAD_PRECISION={on_off(self._current_precision == QUAD)}"
+            ]
+            cmake.configure(cli_args=cli_args)
```

Comment:
```suggestion
            variables = {
                "ENABLE_FLOAT": on_off(self._current_precision == SINGLE),
                "ENABLE_LONG_DOUBLE": on_off(self._current_precision == LONGDOUBLE),
                "EENABLE_QUAD_PRECISION": on_off(self._current_precision == QUAD)
            }
            cmake.configure(variables=variables)
```

You can let CMake helper do the parsing job instead.

https://docs.conan.io/2/reference/tools/cmake/cmake.html#conan.tools.cmake.cmake.CMake.configure